core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -54,108 +55,111 @@ import shapeless.ops.hlist.Tupler
  *   Http.server.serve(foobar.toService)
  * }}}
  */
-trait Endpoint[A] { self =>
+trait Endpoint[F[_], A] { self =>
 
   /**
-   * Request item (part) that's this endpoint work with.
-   */
+    * Request item (part) that's this endpoint work with.
+    */
   def item: items.RequestItem = items.MultipleItems
 
   /**
-   * Runs this endpoint.
-   */
-  def apply(input: Input): Endpoint.Result[A]
+    * Runs this endpoint.
+    */
+  def apply(input: Input): EndpointResult[F, A]
 
   /**
-   * Maps this endpoint to the given function `A => B`.
-   */
-  final def map[B](fn: A => B): Endpoint[B] =
-    mapAsync(fn.andThen(Future.value))
+    * Maps this endpoint to the given function `A => B`.
+    */
+  final def map[B](fn: A => B)(implicit F: Monad[F]): Endpoint[F, B] =
```

Comment:
I see we require deferent instances depending on what we do. I'd assume your reasoning was to require the least-powerful abstraction suitable for a job. I like this in general, but maybe we should just fix everything into `Effect` and stop worrying about this?

I will try to see what others doing, but if you have some insights on why having different effects could be helpful, please share.

Reply:
I think it's a matter of approach, but i.e. testing & calling some separate methods sometimes doesn't require you to have `Effect` instance, so it's possible to use any valid `F[_]` such as `Try`

Reply:
That's a good reason to use least powerful typeclass. Let's keep it as is for now. We can always revisit later if we decide that sticking to `Effect` everywhere is a better approach.